hypothesis-python/src/hypothesis/core.py

```diff
@@ -1477,6 +1491,9 @@ def find(
     )
 
     if database_key is None and settings.database is not None:
+        # Note: The database key is not guaranteed to be unique. If not, replaying
+        # of database examples may fail to reproduce or overrun (raise
+        # HealthCheck.data_too_large) due to being replayed on the wrong condition.
```

Comment:
IMO that replaying examples from the database can raise a healthcheck is clearly a bug on our end, if we can get a short repro that would be great - even with hand-written adversarial database contents (using InMemoryDataBase) and low-level conjecture draw functions to ensure that we reproducibly trigger the error.

We might want to defer that to another PR though, and get each of the logic pieces in separately / sooner.